### PR TITLE
feat: add Celo staking

### PR DESCRIPTION
**Pull request by andyhass**
Co-authored-by: Paweł Nguyen <pawel.nguyen@gmail.com>

### 📝 Description

This PR adds the ability to stake Celo 🙌 

Note: While it is possible to view staking operations in Ledger Live Mobile, the ability to stake Celo on LLM does not exist yet.  As of now, we are planning on including that functionality in a later release.

### ❓ Context

- **Impacted projects**: Ledger Live Desktop, Ledger Live Mobile, Ledger Live Common, Cryptoassets
- **Linked resource(s)**: Non-applicable

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

[Demo Video](https://www.dropbox.com/s/kbqjclhkwqx7iub/Celo.mp4?dl=0)

### CODE REVIEW

File: apps/ledger-live-desktop/src/config/urls.js [hunk 1]
```diff
@@ -218,13 +218,23 @@ export const urls = {
   elrond: {
     website: "https://elrond.com",
   },
+  figment: {
+    website: "https://www.figment.io",
+  },
   solana: {
     staking: "https://support.ledger.com/hc/en-us/articles/4731749170461?docs=true",
     recipient_info: "https://support.ledger.com",
-    ledgerByFigmentTC: "https://drive.google.com/file/d/1vlIh2gTwtbMon8_bzFQGjCLhTUqS5uQc",
```
Comment (andyhass) [inline apps/ledger-live-desktop/src/config/urls.js:224]: ☝️ Fixing the link to Figment's terms and conditions 👇

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 2]
```diff
@@ -573,10 +573,6 @@ const cryptocurrenciesById: Record<string, CryptoCurrency> = {
       },
     ],
     explorerViews: [
```
Comment (andyhass) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:575]: 👇 Removing the explorer link to Hubble - it's a deprecated service.  `https://explorer.celo.org` is a fine default explorer.

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 3]
```diff
@@ -2097,6 +2103,270 @@
       "MERGE_OUTPUTS": "Minimize future fees (merge coins)"
     }
   },
```
Comment (andyhass) [inline apps/ledger-live-desktop/static/i18n/en/app.json:2105]: Ledger team - is there anything else I need to do for the other localizations?

File: libs/ledger-live-common/src/apps/support.ts [hunk 4]
```diff
@@ -35,6 +35,7 @@ const appVersionsRequired = {
   Elrond: ">= 1.0.11",
   Ethereum: ">= 1.9.17",
   Solana: ">= 1.2.0",
```
Comment (andyhass) [inline libs/ledger-live-common/src/apps/support.ts:37]: 👇 For staking, at least version 1.1.8 of the Nano app is required.

**Review by aelmilligy-ledger — COMMENTED** [review]
Hi, 
Could you add the file bridge.integration.test.ts ? 
To help you: 

- Here an example: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/families/algorand/bridge.integration.test.ts 

- And a link to our developer portal (our documentation) : https://developers.ledger.com/docs/coin/writing-tests/#writing-bridgeintegrationtestts

File: apps/ledger-live-desktop/static/i18n/en/app.json (hunk 3 above)
Comment (haammar-ledger) [reply apps/ledger-live-desktop/static/i18n/en/app.json:2105]: Hey @andyhass nope that's perfect, only English is required, translations are automatically provided afterwards by a dedicatd team. :slightly_smiling_face:

**Review by haammar-ledger — COMMENTED** [review]
Awesome work, thanks!

To clarify, is it planned that the mobile implementation is part of this PR?
If not, need to test the behavior on mobile and make sure it's not broken.

**Review by haammar-ledger — CHANGES_REQUESTED** [review]
Also can you please add tests?

There are two major test coverage that we require in the live-common part:
- `bridge.integration.test.ts` as mentioned by @aelmilligy-ledger : see here: https://developers.ledger.com/docs/coin/writing-tests/#writing-bridgeintegrationtestts
- `specs.ts` for the bot tests, see here: https://developers.ledger.com/docs/coin/writing-tests/#testing-the-transaction-broadcast-with-the-bot

Also, we have recently introduced the Playwright framework, for end-to-end testing. It's not yet documented in the developer portal but all info is available here: https://github.com/LedgerHQ/ledger-live/wiki/LLD:Testing
Since you are adding a lot of UI it would be very nice to have some coverage there too, if you can.

Thank you!

File: apps/ledger-live-desktop/static/i18n/en/app.json (hunk 3 above)
Comment (haammar-ledger) [reply apps/ledger-live-desktop/static/i18n/en/app.json:2105]: Hey @andyhass nope that's perfect, only English is required, translations are automatically provided afterwards by a dedicated team. slightly_smiling_face

File: libs/ledger-live-common/src/families/celo/types.ts [hunk 5]
```diff
@@ -1,20 +1,99 @@
...
+  electionAddress: string | null | undefined;
+  lockedGoldAddress: string | null | undefined;
+};
 export type Transaction = TransactionCommon & {
   family: "celo";
   fees: BigNumber | null | undefined;
+  mode: CeloOperationMode;
+  index: number | null | undefined;
+  address: string | null | undefined;
+  group: string | null | undefined;
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/celo/types.ts:58]: _address_ and _group_ are very vague, I suggest more specific field names, just to ease readability and understanding.

File: apps/ledger-live-desktop/src/renderer/families/celo/utils.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  // account's registration status is false, then
+  // we know that the account is truly not registered yet.
+  const account = accounts.find(currentAccount => accountId === currentAccount.id);
+  const isAccountRegistrationPending =
+    !!account &&
+    account.pendingOperations.some(currentAccount => currentAccount.type === "REGISTER") &&
+    !account.celoResources.registrationStatus;
+
+  return isAccountRegistrationPending;
+};
```
[the remark this answers is not in the document]
Comment (andyhass) [reply apps/ledger-live-desktop/src/renderer/families/celo/utils.js]: You're right @haammar-ledger - I did move this function to the `logic.ts` file.  Thanks!

File: libs/ledger-live-common/src/families/celo/types.ts (hunk 5 above)
Comment (andyhass) [reply libs/ledger-live-common/src/families/celo/types.ts]: As a matter of fact, I was able to remove these - they were unnecessary.

**Review by haammar-ledger — APPROVED** [review]
Great job, thanks!

**Pull request merged**